File: Creational/FactoryMethod/Python/Structure/product.py
"""
Product interface - defines the interface for objects the factory method creates
"""
from abc import ABC, abstractmethod


class Product(ABC):
    """
    Abstract Product class - defines the interface that all concrete products must implement
    """
    
    @abstractmethod
    def operation(self) -> str:
        """
        Define an operation that all concrete products must implement
        
        Returns:
            str: Description of the product's operation
        """
        pass

File: Creational/FactoryMethod/Java/Structure/Product.java
/**
 * Product interface - định nghĩa interface chung cho tất cả products
 * được tạo ra bởi factory method
 */
public interface Product {
    /**
     * Phương thức chung mà tất cả concrete products phải implement
     */
    void operation();
    
    /**
     * Trả về tên của product
     */
    String getName();
}
